sources/TileDB.CSharp/Marshalling/MarshaledStringOut.cs

```diff
@@ -11,6 +12,16 @@ internal static class MarshaledStringOut
         public static string GetString(ReadOnlySpan<byte> span) =>
             Encoding.ASCII.GetString(span);
 
+        public static string GetString(ReadOnlySpan<byte> span, DataType dataType) =>
+            dataType switch
+            {
+                DataType.StringAscii => Encoding.ASCII.GetString(span),
+                DataType.StringUtf8 => Encoding.UTF8.GetString(span),
+                DataType.StringUtf16 => Encoding.Unicode.GetString(span),
+                DataType.StringUtf32=> Encoding.UTF32.GetString(span),
+                _ => throw new ArgumentOutOfRangeException(nameof(dataType), dataType, "Unsupported string data type.")
```

Comment:
I didn't handle UCS-2 and UCS-4; they are legacy, don't have framework support and need custom code to convert. Is that OK?

Reply:
Yes